docs: ui capabilities text area PTL-1580 (#1900)
genesiscommunitysuccess/docs: docs/001_develop/03_client-capabilities/008_interaction/text-area.mdx

## API

### Attributes:

<table>
  <thead><tr><th>Name</th><th>Type</th><th>Description</th><th>Example</th></tr></thead>
  <tbody>
    <tr>
      <td>appearance</td>

>       <td><code>string</code></td>

Reviewer: ```suggestion
      <td><code>"filled" | "outline"</code></td>
```

<td>

```typescript
<text-area formId="formId">
```

</td>
    </tr>
    <tr>
      <td>maxlength</td>
      <td><code>number</code></td>
      <td>The maximum number of characters allowed</td>
<td>

```typescript
<text-area maxlength=100>
```

</td>
    </tr>
    <tr>
      <td>minlength</td>
      <td><code>number</code></td>
      <td>The minimum number of characters allowed</td>
<td>

```typescript
<text-area minlength=50>
```

</td>
    </tr>
    <tr>
      <td>name</td>
      <td><code>string</code></td>
      <td>Defines the name of the element</td>
<td>

```typescript
<text-area name="textAreaName">
```

</td>
    </tr>
    <tr>
      <td>placeholder</td>
      <td><code>string</code></td>
      <td>Sets placeholder text for the element, which disappears when the user starts inputting</td>
<td>

```typescript
<text-area placeholder="Sample place holder text">
```

</td>
    </tr>
    <tr>
      <td>readonly</td>
      <td><code>boolean</code></td>
      <td>If true, the user cannot change the value of this field</td>
<td>

```typescript
<text-area readonly>
```

</td>
    </tr>
    <tr>
      <td>resize</td>

>       <td><code>string</code></td>
>       <td>Allows the user to resize the element; this can be {"none","both","vertical","horizontal"}. **Default: none**</td>

Reviewer: This is the same issue I had, the `{ ... }` syntax you're using doesn't render correctly on docusouras. 
```suggestion
      <td><code>"none" | "both" | "vertical" | "horizontal"</code></td>
      <td>Configures which ways the user can resize the text area. **Default: none**</td>
```

![Screenshot 2024-11-18 at 08 39 15](https://github.com/user-attachments/assets/1426b010-61e6-44b5-834e-871109d7ec0f)

<td>

```typescript
<text-area resize="vertical">
```

</td>
    </tr>
    <tr>
      <td>rows</td>
      <td><code>number</code></td>
      <td>Defines the size of the element vertically. **Default: 2**</td>
<td>

```typescript
<text-area rows=5>
```

</td>
    </tr>
    <tr>
      <td>value</td>
      <td><code>string</code></td>

>       <td>The text area content</td>
> <td>
> 
> ```typescript
> <text-area value="Text area value">
> ```

Reviewer: I think attributes like this are useful to show using sync as it's unlikely the user will just set the value on the attribute.
```suggestion
      <td>The text area content. Use the `sync` utility to bind the data from the template back to the model.</td>
<td>

```typescript
<text-area value=${sync((x) => x.textAreaValue)}>
```
```

</td>
    </tr>
  </tbody>
</table>

> 

Reviewer: ```suggestion

### Properties

This component doesn't have any properties which are not also controlled via attributes.

```
